src/ray/raylet/scheduling/cluster_task_manager.cc

```diff
@@ -184,6 +185,74 @@ bool ClusterTaskManager::CancelTask(const TaskID &task_id) {
   return false;
 }
 
+void ClusterTaskManager::Heartbeat(std::shared_ptr<HeartbeatTableData> data,
+                                   bool light_heartbeat_enabled) const {
+  auto resource_loads = data->mutable_resource_load();
+  auto resource_load_by_shape =
+      data->mutable_resource_load_by_shape()->mutable_resource_demands();
+
+  if (light_heartbeat_enabled) {
+    RAY_CHECK(false) << "TODO";
+  } else {
+    // TODO (Alex): Implement the 1-CPU task optimization.
+    for (auto it = tasks_to_schedule_.begin(); it != tasks_to_schedule_.end(); it++) {
+      auto &task = std::get<0>(*it);
+      auto &resources =
+          task.GetTaskSpecification().GetRequiredResources().GetResourceMap();
+
+      auto by_shape_entry = resource_load_by_shape->Add();
+
+      for (auto to_add_it = resources.begin(); to_add_it != resources.end();
```

Comment:
Also use advanced syntax

Reply:
sorry, what's advanced syntax?

Reply:
Ah, just use `for (const auto to_add_it : resources)`